src/lib/fs/path.c

```diff
@@ -293,3 +341,389 @@ make_path_absolute(char *fname)
   return absfname;
 #endif /* defined(_WIN32) */
 }
+
+/**** Helper functions of type glob_predicate_fn used for glob processing ****/
+
+/** Returns true if <b>file_type</b> represents an existing file (even if
+ * empty). Returns false otherwise. */
+static bool
+is_file(file_status_t file_type, const char *pattern_after)
+{
+  (void)pattern_after;
+  return file_type != FN_ERROR && file_type != FN_NOENT && file_type != FN_DIR;
+}
+
+/** Returns true if <b>file_type</b> represents an existing directory. Returns
+ * false otherwise. */
+static bool
+is_dir(file_status_t file_type, const char *pattern_after)
```

Comment:
The is_dir() function also belongs in files.h.

I'm not sure if it's worth refactoring existing checks to use it.

Reply:
Suggestion applied in ae426f9